### ISSUE TITLE
GRIDSS assembly error

### ISSUE THREAD

**Issue by rdmorin** [LCR-BCCRC/lcr-modules#105]
I ran the full gridss pipeline yesterday on this sample and it completed successfully. I'm trying it again (same inputs) and now it is failing. The file referred to by Snakemake (gridss.log) is empty. The log files in the working directory created for gridss have errors aplenty (below). Does gridss just do this sometimes? Is there a workaround?

```
Error in rule _gridss_paired:
    jobid: 8
    output: results/gambl/gridss-testing-default/02-gridss/genome--grch37/14-11247T--14-11247N--matched/gridss_raw.vcf.gz, results/gambl/gridss-testing-default/02-gridss/genome--grch37/14-11247T--14-11247N--matched/assembly.bam
    log: results/gambl/gridss-testing-default/logs/launched-2020-09-11-at-15-06-34/02-gridss/genome--grch37/14-11247T--14-11247N--matched/gridss.log (check log file(s) for error message)
    conda-env: /projects/rmorin_scratch/conda_environments/b3e7b470
    shell:
        gridss --reference results/gambl/gridss-testing-default/00-inputs/references/grch37/genome_fa/genome.fa --output results/gambl/gridss-testing-default/02-gridss/genome--grch37/14-11247T--14-11247N--matched/gridss_raw.vcf.gz --workingdir `dirname results/gambl/gridss-testing-default/02-gridss/genome--grch37/14-11247T--14-11247N--matched/gridss_raw.vcf.gz` --assembly results/gambl/gridss-testing-default/02-gridss/genome--grch37/14-11247T--14-11247N--matched/assembly.bam --blacklist /projects/rmorin/projects/gambl-repos/gambl-rmorin/ref/lcr-modules-references/genomes/grch37/encode/encode-blacklist.grch37.bed --repeatmaskerbed /projects/rmorin/projects/gambl-repos/gambl-rmorin/ref/lcr-modules-references/genomes/grch37/repeatmasker/repeatmasker.grch37.bed --threads 8 --jvmheap 30000m --labels "14-11247N,14-11247T" --steps assemble,call --picardoptions VALIDATION_STRINGENCY=SILENT results/gambl/gridss-testing-default/00-inputs/bam/genome--grch37/14-11247N.bam results/gambl/gridss-testing-default/00-inputs/bam/genome--grch37/14-11247T.bam > results/gambl/gridss-testing-default/logs/launched-2020-09-11-at-15-06-34/02-gridss/genome--grch37/14-11247T--14-11247N--matched/gridss.log
        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)
```

```
INFO	2020-09-11 15:07:59	PositionalAssembler	Error during assembly of chromosome 7 (96 reads in graph). Attempting recovery by rebuilding assembly graph.
htsjdk.samtools.util.RuntimeIOException: java.util.zip.DataFormatException: invalid distance too far back
	at htsjdk.samtools.util.BlockGunzipper.unzipBlock(BlockGunzipper.java:161)
	at htsjdk.samtools.util.BlockGunzipper.unzipBlock(BlockGunzipper.java:96)
	at htsjdk.samtools.util.BlockCompressedInputStream$CompressionBlock.decompress(BlockCompressedInputStream.java:682)
	at htsjdk.samtools.util.AsyncBlockCompressedInputStream$AsyncBlockCompressedInputStreamTaskRunner.transform(AsyncBlockCompressedInputStream.java:140)
	at htsjdk.samtools.util.AsyncBlockCompressedInputStream$AsyncBlockCompressedInputStreamTaskRunner.transform(AsyncBlockCompressedInputStream.java:119)
	at htsjdk.samtools.util.AsyncReadTaskRunner.processNextBatch(AsyncReadTaskRunner.java:257)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:642)
	at java.base/java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:478)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.util.zip.DataFormatException: invalid distance too far back
	at java.base/java.util.zip.Inflater.inflateBytesBytes(Native Method)
	at java.base/java.util.zip.Inflater.inflate(Inflater.java:385)
	at htsjdk.samtools.util.BlockGunzipper.unzipBlock(BlockGunzipper.java:145)
	... 10 more
```

**Comment by rdmorin** [comment]
@lkhilton I have yet to reproduce this error, but in trying to track down the issue I realized there is probably a bug that crept in[ a very late commit.](https://github.com/LCR-BCCRC/lcr-modules/commit/eb923a238836feb3379031801056ecb5e2616ba3) 
 This probably explains why I had an empty log file. I hadn't even notice this change when reviewing the PR and if I had I probably wouldn't have seen it as a problem but I'll refer you to the comment I added on the commit for details.

**Comment by rdmorin** [comment]
@lkhilton when you have some time can you confirm if this version is working? Note the commit I refer to above.

**Comment by lkhilton** [comment]
I'm running a version of the pipeline that fixes the log file problem. I haven't seen this assembly error come up yet (even on FFPE genomes taking > 24 hours to run). I'll close this when I submit a PR with my updated pipeline.

**Comment by lkhilton** [comment]
Addressed with #117
